zip-0227.rst

```diff
@@ -96,9 +106,47 @@ Issuance Key Derivation
 ```````````````````````
 
 The issuance master key is generated by choosing a bit sequence uniformly at random from :math:`\mathbb{B}^{\mathbb{Y}[32]}`, like the Orchard spending key [#protocol-orchardkeycomponents]_.
-In :math:`\mathsf{zcashd}`, this key is derived in a manner similar to the generation of the Orchard master key in ZIP 32 [#zip-0032-orchard-master]_, as detailed below:
 
-- Details TBD
+Issuance master key derivation for hierarchical deterministic wallets
+''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''' 
+
+In ``zcashd``, this key is derived in a manner similar to the generation of the Orchard master key in ZIP 32 [#zip-0032-orchard-master]_, as detailed in this section.
+
+Let :math:`S` be a seed byte sequence of a chosen length, which MUST be at least 32 bytes and at most 252 bytes.
+
+- Calculate :math:`I = \textsf{BLAKE2b-512}(\texttt{"ZIP32ZSAIssuance"},S)`.
+- Split :math:`I` into two 32-byte sequences :math:`I_L` and :math:`I_R`.
+- Return :math:`(\mathsf{sk}_m, \mathsf{c}_m) := (I_L, I_R)` as the master extended issuance key, :math:`m_{\mathsf{Issuance}}`.
+
+As in ZIP 32 for Orchard [#zip-0032-orchard-child-key-derivation]_, we only use hardened child key derivation for issuance master key. 
+We define the following function for child key derivation:
+
+:math:`\mathsf{CDKik}((\mathsf{sk}_{par},\mathsf{c}_{par}),i) \to (\mathsf{sk}_i, \mathsf{c}_i)`
+
+- Check whether :math:`i \geq 2^{31}` (i.e. whether the child is a hardened key)
+
+  - If so (hardened child): Let :math:`I = \mathsf{PRF^{expand}}(\mathsf{c}_{par}, [\mathtt{0xa1}] \| \mathsf{sk}_{par} \| \mathsf{I2LEOSP}_{32}(i))`.
+  - If not (non-hardened child): return failure.
+
+- Split :math:`I` into two 32-byte sequences :math:`I_L` and :math:`I_R`.
+- Set :math:`(\mathsf{sk}_i, \mathsf{c}_i) = (I_L, I_R)`.
+
+We adapt the notation of ZIP 32 [#zip-0032]_ for shielded HD paths, using prime marks (:math:`'`) to
+indicate hardened derivation (:math:`i' = i + 2^{31}`) as in BIP 44 [#bip-0044]_:
+
+- :math:`\mathsf{CDKik}(\mathsf{CDKik}(\mathsf{CDKik}(m_\mathsf{Issuance}, a), b), c)` is written as :math:`m_\mathsf{Issuance} / a / b / c`.
+
+We define the issuance master key path as :math:`m_\mathsf{Issuance} / purpose' / coin\_type' / account'`. We fix the path levels as follows:
+
+- :math:`purpose`: a constant set to :math:`227'` (or :math:`\texttt{0x800000e3}`) following the BIP 43 recommendation.
+- :math:`coin\_type`: a constant identifying the cryptocurrency that this subtree's keys are used with. We use the same constants as defined in SLIP 44
+  [#slip-0044]_ for ZEC, namely :math:`\texttt{0x80000085}`. Also, in keeping with that document, all cryptocurrency testnets share :math:`coin\_type` index :math:`1`.
```

Comment:
I explicitly fixed it to the ZEC value, because if we write it the way it is written in ZIP 32, it could be taken to mean that if we are issuing a wrapped BTC we should use the `coin_type` for BTC, and so on.

I think fixing it to ZEC seems to be the best approach, but suggestions are welcome.